### PR TITLE
Exercise for New contributors

### PR DESCRIPTION
**Pull request by moadmmh**
I have just finished the given exercise for new contributors mentioned [ here ](https://gcompris.net/wiki/An_exercise_for_new_contributors)

### CODE REVIEW

File: src/activities/moadmmh_activity/ActivityInfo.qml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ActivityInfo {
+  name: "moadmmh_activity/Moadmmh_activity.qml"
+  difficulty: 1
+  icon: "moadmmh_activity/moadmmh_activity.svg"
+  author: "Moad mohammed Elhebri &lt;moadmohammedelhebri@gmail.com&gt;"
+  demo: true
+  //: Activity title
+  title: "Moadmmh Activity"
+  //: Help title
+  description: "This Activity is part of the first contribution exercise"
```
Comment (petitlapin) [inline src/activities/moadmmh_activity/ActivityInfo.qml]: To be able to translate strings, we need qsTr("...")

File: src/activities/moadmmh_activity/Moadmmh_activity.qml [hunk 2]
```diff
@@ -0,0 +1,130 @@
+/* GCompris - moadmmh_activity.qml
+ *
+ * Copyright (C) 2018 YOUR NAME <xx@yy.org>
```
Comment (petitlapin) [inline src/activities/moadmmh_activity/Moadmmh_activity.qml]: don't forget to update this too if you create files

File: src/activities/moadmmh_activity/Moadmmh_activity.qml [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+ *
+ * Authors:
+ *   <THE GTK VERSION AUTHOR> (GTK+ version)
```
Comment (petitlapin) [inline src/activities/moadmmh_activity/Moadmmh_activity.qml]: if it is not a port, you can remove it

File: src/activities/moadmmh_activity/Moadmmh_activity.qml [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+            MouseArea {
+                anchors.fill: parent
+                onPressed: rectangle.state = "PRESSED"
+                onReleased: rectangle.state = "RELEASED"
+            }
+
+            states: [
+                State {
+                    name: "PRESSED"
+                    PropertyChanges { target: rectangle; color: "blue"}
```
Comment (petitlapin) [inline src/activities/moadmmh_activity/Moadmmh_activity.qml]: missing space at the end before }

File: src/activities/moadmmh_activity/Moadmmh_activity.qml [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+            property alias bar: bar
+            property alias bonus: bonus
+        }
+
+        onStart: { Activity.start(items) }
+        onStop: { Activity.stop() }
+
+        Rectangle {
+            id: rectangle
+            width: 300; height: 300
```
Comment (petitlapin) [inline src/activities/moadmmh_activity/Moadmmh_activity.qml]: be careful to not hardcode dimensions/positions/sizes as GCompris is available on multiple different supports like phones, tablets and computers.

File: src/activities/moadmmh_activity/Moadmmh_activity.qml [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+                    from: "RELEASED"
+                    to: "PRESSED"
+                    ColorAnimation { target: rectangle; duration: 100}
+                }
+            ]
+        }
+
+        GCText {
+            anchors.centerIn: parent
+            text: "An Exercise Activity Done by Moad Mohammed Elhebri"
```
Comment (petitlapin) [inline src/activities/moadmmh_activity/Moadmmh_activity.qml]: qsTr()

File: src/activities/moadmmh_activity/Moadmmh_activity.qml [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+                },
+                Transition {
+                    from: "RELEASED"
+                    to: "PRESSED"
+                    ColorAnimation { target: rectangle; duration: 100}
+                }
+            ]
+        }
+
+        GCText {
```
Comment (petitlapin) [inline src/activities/moadmmh_activity/Moadmmh_activity.qml]: good use of GCText, we need the text to all have the same font

**Comment by moadmmh** [comment]
Okey i will keep all of the above comments in mind thank you again for your review

**Comment by moadmmh** [comment]
Hello i was just wondering if i take your comments in consideration for real commits or i have to modify the code for the contributor exercise  and make another pull request ?

**Comment by petitlapin** [comment]
hi, only for real commits :), no need to update this PR

**Pull request closed without merging**